### PR TITLE
Remove `zero(::UnitQuaternion)`

### LINKED ISSUE #157: `zero(UnitQuaternion)` doesn't make sense (for me)
Here's an example script:
```julia
julia> using Rotations

julia> zero(UnitQuaternion)
3×3 UnitQuaternion{Float64} with indices SOneTo(3)×SOneTo(3)(1.0, 0.0, 0.0, 0.0):
 1.0  0.0  0.0
 0.0  1.0  0.0
 0.0  0.0  1.0

julia> zero(UnitQuaternion) + zero(UnitQuaternion)
3×3 StaticArrays.SMatrix{3, 3, Float64, 9} with indices SOneTo(3)×SOneTo(3):
 2.0  0.0  0.0
 0.0  2.0  0.0
 0.0  0.0  2.0

julia> zero(UnitQuaternion) + zero(UnitQuaternion) == zero(UnitQuaternion)
false

julia> zero(UnitQuaternion) == one(UnitQuaternion)
true
```
And we don't have tests for `zero(UnitQuaternion)`, so maybe we can just drop the definition.
https://github.com/JuliaGeometry/Rotations.jl/blob/v1.0.2/src/unitquaternion.jl#L204

### PR DESCRIPTION
**Pull request by hyrodium**
This PR fixes #157.

### CODE REVIEW

**Comment by hyrodium** [comment]
I've also added `ArgumentError` to avoid `zero(RotMatrix3{Float64})`.

### Before this PR
```julia
julia> zero(RotMatrix3{Float64})
3×3 RotMatrix3{Float64} with indices SOneTo(3)×SOneTo(3):
 0.0  0.0  0.0
 0.0  0.0  0.0
 0.0  0.0  0.0
```

### After this PR
```julia
julia> zero(RotMatrix3{Float64})
ERROR: ArgumentError: There is no additive identity for Rotation. Consider using the one(RotMatrix3{Float64}) method instead.
Stacktrace:
 [1] zero(T::Type{RotMatrix3{Float64}})
   @ Rotations ~/.julia/dev/Rotations/src/core_types.jl:14
 [2] top-level scope
   @ REPL[2]:1
```

File: src/core_types.jl [hunk 1]
```diff
@@ -11,6 +11,8 @@ Base.@pure StaticArrays.Size(::Type{Rotation{N,T}}) where {N,T} = Size(N,N)
 Base.@pure StaticArrays.Size(::Type{R}) where {R<:Rotation} = Size(supertype(R))
 Base.adjoint(r::Rotation) = inv(r)
 Base.transpose(r::Rotation{N,T}) where {N,T<:Real} = inv(r)
+Base.zero(T::Type{<:Rotation}) = throw(ArgumentError("There is no additive identity for Rotation. Consider using the one($T) method instead."))
+Base.zero(r::Rotation) = throw(ArgumentError("There is no additive identity for Rotation. Consider using the one($(typeof(r))($r)) method instead."))
```
Comment (goretkin) [inline src/core_types.jl:15]: See comment at https://github.com/JuliaGeometry/Rotations.jl/issues/157#issuecomment-844294269

Comment (hyrodium) [reply src/core_types.jl:15]: Here's an updated error message:
```julia
"$T is a subtype of AbstractMatrix, but there is no additive identity for Rotation. To get a non-rotation instance, consider using the one($T) method instead."
```
Is there any problem with this? I'm not native English speaker, so there may be some mistakes. :)

**Comment by andyferris** [comment]
Thanks @hyrodium.

I'm a bit concerned about the argument error, though. Every rotation is meant to be a fully-functioning `AbstractMatrix`. Generic code might call `zero(x)`, in which case we can just return an `SMatrix`. Just like `rotation + rotation` gives an `SMatrix` (you can think of `zero()` providing the identity for the `+` operation, which is defined but not closed).

Currently we have:

```julia
julia> r = one(RotMatrix3{Float64})
3×3 RotMatrix3{Float64} with indices SOneTo(3)×SOneTo(3):
 1.0  0.0  0.0
 0.0  1.0  0.0
 0.0  0.0  1.0

julia> r + r
3×3 StaticArrays.SMatrix{3, 3, Float64, 9} with indices SOneTo(3)×SOneTo(3):
 2.0  0.0  0.0
 0.0  2.0  0.0
 0.0  0.0  2.0

julia> zero(r)
3×3 RotMatrix3{Float64} with indices SOneTo(3)×SOneTo(3):
 0.0  0.0  0.0
 0.0  0.0  0.0
 0.0  0.0  0.0
```

We are extending the `AbstractArray` interface here - as a general rule, we can add functionality but not subtract functionality.

**Comment by hyrodium** [comment]
I've updated the `zero` funciton.

```julia
julia> zero(MRP)
3×3 StaticArrays.SMatrix{3, 3, Float64, 9} with indices SOneTo(3)×SOneTo(3):
 0.0  0.0  0.0
 0.0  0.0  0.0
 0.0  0.0  0.0

julia> zero(RotMatrix{3})
3×3 StaticArrays.SMatrix{3, 3, Float64, 9} with indices SOneTo(3)×SOneTo(3):
 0.0  0.0  0.0
 0.0  0.0  0.0
 0.0  0.0  0.0

julia> zero(RotMatrix{2})
2×2 StaticArrays.SMatrix{2, 2, Float64, 4} with indices SOneTo(2)×SOneTo(2):
 0.0  0.0
 0.0  0.0

julia> zero(Angle2d)
2×2 StaticArrays.SMatrix{2, 2, Float64, 4} with indices SOneTo(2)×SOneTo(2):
 0.0  0.0
 0.0  0.0
```

But, there sitll be some issues around `zeros` funciton.

```julia
julia> zeros(MRP,2)
2-element Vector{MRP}:
 [1.0 0.0 0.0; 0.0 1.0 0.0; 0.0 0.0 1.0]
 [1.0 0.0 0.0; 0.0 1.0 0.0; 0.0 0.0 1.0]

julia> zeros(RotMatrix{3},2)
2-element Vector{RotMatrix{3, T, L} where {T, L}}:
 [0.0 0.0 0.0; 0.0 0.0 0.0; 0.0 0.0 0.0]
 [0.0 0.0 0.0; 0.0 0.0 0.0; 0.0 0.0 0.0]

julia> zeros(RotMatrix{2},2)
2-element Vector{RotMatrix{2, T, L} where {T, L}}:
 [0.0 0.0; 0.0 0.0]
 [0.0 0.0; 0.0 0.0]

julia> zeros(Angle2d,2)
2-element Vector{Angle2d}:
Error showing value of type Vector{Angle2d}:
ERROR: MethodError: no method matching cos(::NTuple{4, Float64})
```

I'll fix them after tomorrow.

**Comment by andyferris** [comment]
Awesome.

> At first, I thought zero(T) should be a instance of T

There was an old discussion at julialang/julia about this once... here's a recent one quoting the docs: https://github.com/JuliaLang/julia/issues/40796#issuecomment-839007123

Basically we want things like `isequal(x, x + zero(x))` and `isequal(x, x * one(x))`, but not always that `zero(x) isa typeof(x)` or `one(x) isa typeof(x)`.

**Comment by hyrodium** [comment]
I've updated `zeros` method:
```julia
julia> zeros(MRP,2)
2-element Vector{StaticArrays.SMatrix{3, 3, Float64, 9}}:
 [0.0 0.0 0.0; 0.0 0.0 0.0; 0.0 0.0 0.0]
 [0.0 0.0 0.0; 0.0 0.0 0.0; 0.0 0.0 0.0]

julia> zeros(RotMatrix{3},2)
2-element Vector{StaticArrays.SMatrix{3, 3, Float64, 9}}:
 [0.0 0.0 0.0; 0.0 0.0 0.0; 0.0 0.0 0.0]
 [0.0 0.0 0.0; 0.0 0.0 0.0; 0.0 0.0 0.0]

julia> zeros(RotMatrix{2},2)
2-element Vector{StaticArrays.SMatrix{2, 2, Float64, 4}}:
 [0.0 0.0; 0.0 0.0]
 [0.0 0.0; 0.0 0.0]

julia> zeros(Angle2d,2)
2-element Vector{StaticArrays.SMatrix{2, 2, Float64, 4}}:
 [0.0 0.0; 0.0 0.0]
 [0.0 0.0; 0.0 0.0]
```

Usually, `zeros(T,n)` returns a instance of `Vector{T}`. (e.g. `zeros(Real, 3)`)
But, `zeros(MRP,2)` can't be `Vector{MRP}` because `zero(MRP) <: MRP` is `false`.

**Comment by hyrodium** [comment]
>There was an old discussion at julialang/julia about this once... here's a recent one quoting the docs: JuliaLang/julia#40796 (comment)
>
>Basically we want things like isequal(x, x + zero(x)) and isequal(x, x * one(x)), but not always that zero(x) isa typeof(x) or one(x) isa typeof(x).

Thanks for the information!

**Comment by hyrodium** [comment]
bump :)

**Pull request merged**